docs/guide/structure-controllers.md

```diff
@@ -112,32 +112,32 @@ For this reason, controller IDs are often nouns referring to the types of the re
 For example, you may use `article` as the ID of a controller that handles article data.
 
 By default, controller IDs should contain these characters only: English letters in lower case, digits,
-underscores, dashes and forward slashes. For example, `article` and `post-comment` are both valid controller IDs,
+underscores, hyphens, and forward slashes. For example, `article` and `post-comment` are both valid controller IDs,
 while `article?`, `PostComment`, `admin\post` are not.
 
 A controller ID may also contain a subdirectory prefix. For example, `admin/article` stands for an `article` controller
 in the `admin` subdirectory under the [[yii\base\Application::controllerNamespace|controller namespace]].
-Valid characters for subdirectory prefixes include: English letters in lower and upper cases, digits, underscores and
+Valid characters for subdirectory prefixes include: English letters in lower and upper cases, digits, underscores, and
 forward slashes, where forward slashes are used as separators for multi-level subdirectories (e.g. `panels/admin`).
 
 
 ### Controller Class Naming <span id="controller-class-naming"></span>
 
-Controller class names can be derived from controller IDs according to the following rules:
+Controller class names can be derived from controller IDs according to the following procedure:
 
-* Turn the first letter in each word separated by dashes into upper case. Note that if the controller ID
+1. Turn the first letter in each word separated by hyphens into upper case. Note that if the controller ID
   contains slashes, this rule only applies to the part after the last slash in the ID.
-* Remove dashes and replace any forward slashes with backward slashes.
-* Append the suffix `Controller`.
-* And prepend the [[yii\base\Application::controllerNamespace|controller namespace]].
+1. Remove hyphens and replace any forward slashes with backward slashes.
+1. Append the suffix `Controller`.
+1. Prepend the [[yii\base\Application::controllerNamespace|controller namespace]].
```

Comment:
I fully understand your arguments and they are valid in general. In our case, changes are reviewed by many people so we can go with explicit numbering without the fear to introduce errors.